packages/commonwealth/client/scripts/views/pages/MyCommunityStake/Stakes/Stakes.tsx

```diff
@@ -53,11 +54,40 @@ const columnInfo = [
 const Stakes = ({ filterOptions, addressFilter }: TransactionHistoryProps) => {
   const data = useTransactionHistory({ filterOptions, addressFilter });
 
+  // aggregate transaction per community per address
+  const stakes = (() => {
+    const map = {};
+
+    data.map((x) => {
+      const key = (x.community.id + x.address).toLowerCase();
+      const action = x.action === 'mint' ? 1 : -1;
+
+      map[key] = {
+        ...x,
+        ...(map[key] || {}),
+        stake: (map[key]?.stake || 0) + x?.stake * action,
+        voteWeight: (map[key]?.voteWeight || 0) + x?.voteWeight * action,
+        avgPrice:
+          (map[key]?.avgPrice || 0) +
+          parseFloat(
+            (parseFloat(x.price) / WEI_PER_ETHER / x.stake).toFixed(5),
+          ) *
+            action,
+      };
+    });
+
+    return Object.values(map).map((x: any) => ({
+      ...x,
+      voteWeight: x.voteWeight + 1, // total vote weight is +1 of the stake weight
```

Comment:
we shouldn't manipulate vote weight on the front end. I believe that we have this data on the backend and it should be our source of truth

Reply:
Yep that was an oversight on my part. We currently only track stake transactions that go through our website, so this value might not be correct (If user makes stake transactions from outside our website). We should probably query it from the chain here if feasible.

Reply:
Not sure what are the next steps here? cc: @kurtisassad. From what i understand querying chain directly to get all transactions history might not be fesiable?